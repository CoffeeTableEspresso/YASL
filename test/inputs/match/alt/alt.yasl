

fn f(n) {
    match n {
        [ 0, let x, let y ] | [ 1, let y, let x ] {
            echo "x: #{x}, y: #{y}"
        }
        [ 0, let x, let y, ... ] | [ 1, let y, let x, ... ] {
             echo "x: #{x}, y: #{y}, ..."
        }
        { .t: 0, .y: let y, .x: let x } |  { .t: 1, .x: let x, .y: let y } {
            echo "x: #{x}, y: #{y}"
        }
        { .t: 0, .y: let y, .x: let x, ...} | { .t: 1, .x: let x, .y: let y, ... } {
            echo "x: #{x}, y: #{y}, ..."
        }
        * {
            echo '???'
        }
    }
}

f([ 1, 2, 3 ])
f([ 0, 2, 3 ])

f([ 1, 2, 3, 5 ])
f([ 0, 2, 3, 5 ])

f({.t: 1, .x: 10, .y: 20})
f({.t: 0, .x: 10, .y: 20})

f({.t: 1, .x: 10, .y: 20, .v: 5})
f({.t: 0, .x: 10, .y: 20, .v: 5})
