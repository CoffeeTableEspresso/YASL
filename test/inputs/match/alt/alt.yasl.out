x: 3, y: 2
x: 2, y: 3
x: 3, y: 2, ...
x: 2, y: 3, ...
x: 10, y: 20
x: 10, y: 20
x: 10, y: 20, ...
x: 10, y: 20, ...
